### ISSUE TITLE
conditional model training / validation loss.

### ISSUE THREAD

**Issue by jgallowa07** [matsengrp/multidms#140]
Note to add conditional loss functionality for both `Model` and `ModelCollection` (like these seen in the original manuscript submission).

![Screenshot from 2024-03-05 16-52-06](https://github.com/matsengrp/multidms/assets/23087427/35a8f8fb-6b16-429c-a6ff-2b0530c581ea)

Note this is already being done in the spike analysis topic branch, but wanted to document it here as well.
